web/packages/shared/utils/error.ts

```diff
@@ -0,0 +1,70 @@
+/**
+ * Teleport
+ * Copyright (C) 2025 Gravitational, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+/**
+ * Converts any unknown input into an Error instance.
+ * Preserves message and name if available.
+ */
+export function ensureError(err: unknown): Error {
+  if (err instanceof Error) {
+    return err;
+  }
+
+  if (err === null || err === undefined) {
+    return new Error();
```

Comment:
One thing that comes to my mind: maybe we could always include the original value as the `cause` of the error?